tools/common/Assembly.cs

```diff
@@ -328,6 +328,14 @@ public static LinkWithAttribute GetLinkWithAttribute (CustomAttribute attr)
 			return linkWith;
 		}
 
+		void AddFramework (string file)
+		{
+			if (Driver.GetFrameworks (App).TryGetValue (file, out var framework) && framework.Version > App.SdkVersion)
+				ErrorHelper.Warning (135, "Did not link system framework '{0}' as Xcode SDK '{1}' is too old.", file, App.SdkVersion);
```

Comment:
I didn't realize it when you asked in #mac, but this is now showing less information than before (in particular which assembly caused the warning, which can be quite useful). We also know which SDK version is required (so we can say that too), but this is probably less useful (although I always like to give as much information as possible in error/warning messages, since you never know what will be useful).

Also `Xcode SDK` is not entirely correct, if this is a watchOS project you'll end up with messages like `Did not link system framework 'XYZ' as Xcode SDK 4.0 is too old.`, and people end up confused because Xcode 4.0 is **old**.